### PR TITLE
Make possible to run on Windows

### PR DESCRIPTION
**Pull request by to-bar**
- Ensure SSH key permissions
- Adjust UID and GID of epiuser
- Print environment variables for `apply-azepi`
- Support docker.sock for Git Bash on Windows
- Use epicli v0.8.0

### CODE REVIEW

File: resources/azepi/apply.py [hunk 1]
```diff
@@ -51,6 +53,40 @@ def _run_epicli_apply(v):
...
+
+    epiphany_config_docs = load_yaml(module_config["config"])
+
+    cluster = select(epiphany_config_docs,
+                     q_kind("epiphany-cluster"),
+                     exactly=1)
+
+    key_path_from_spec = cluster["specification"]["admin_user"]["key_path"]
+
+    key_path_from_share = os.path.join(v["shared_dir"], "vms_rsa")
```
Comment (sk4zuzu) [inline resources/azepi/apply.py:69]: Would be nice to keep the convention and use `pathlib.Path` here as well :+1::innocent:
```suggestion
    key_path_from_share = v["shared_dir"] / "vms_rsa"
```

File: resources/azepi/apply.py [hunk 2]
```diff
@@ -51,6 +53,40 @@ def _run_epicli_apply(v):
...
+
+    module_config = load_yaml(v["config_file"])[v["M_MODULE_SHORT"]]
+
+    epiphany_config_docs = load_yaml(module_config["config"])
+
+    cluster = select(epiphany_config_docs,
+                     q_kind("epiphany-cluster"),
+                     exactly=1)
+
+    key_path_from_spec = cluster["specification"]["admin_user"]["key_path"]
```
Comment (sk4zuzu) [inline resources/azepi/apply.py:67]:
```suggestion
    key_path_from_spec = get_path(cluster["specification"]["admin_user"]["key_path"])
```

File: resources/azepi/apply.py [hunk 3]
```diff
@@ -51,6 +53,40 @@ def _run_epicli_apply(v):
...
+
+    key_path_from_share = os.path.join(v["shared_dir"], "vms_rsa")
+
+    command = textwrap.dedent(f'''
+        set -e
+        if [ ! -f "{key_path_from_spec}" ]; then
+            echo "File {key_path_from_spec} not found, will copy {key_path_from_share}"
+            cp --verbose {key_path_from_share}* {os.path.dirname(key_path_from_spec)}
```
Comment (sk4zuzu) [inline resources/azepi/apply.py:75]:
```suggestion
            cp --verbose {key_path_from_share}* {key_path_from_spec.parent)}
```

File: resources/azepi/apply.py [hunk 4]
```diff
@@ -1,7 +1,9 @@
 """Implementation of the "apply" method."""
 
+import os
```
Comment (sk4zuzu) [inline resources/azepi/apply.py:3]:
```suggestion
```

**Review by sk4zuzu — CHANGES_REQUESTED** [review]
:heart_eyes:

**Review by sk4zuzu — DISMISSED** [review]
:hugs:

**Review by sk4zuzu — DISMISSED** [review]
:cat2:

File: tests/integration/test_init_minimal_cluster_with_applications.py [hunk 5]
```diff
@@ -128,8 +128,13 @@ def test_init_minimal_cluster_with_applications():
         with (shared_dir / "state.yml").open("w") as stream:
             stream.write(STATE_FILE_MOCK)
 
-        external_cache_dir = pathlib.Path(
-            os.getenv("CACHE_DIR", "/shared")).resolve()
+        cache_dir = os.getenv("CACHE_DIR", "/shared")
```
Comment (sk4zuzu) [inline tests/integration/test_init_minimal_cluster_with_applications.py:131]:
```suggestion
        cache_dir_env = os.getenv("CACHE_DIR", "/shared")
```

File: tests/integration/test_init_minimal_cluster_with_applications.py [hunk 6]
```diff
@@ -128,8 +128,13 @@ def test_init_minimal_cluster_with_applications():
...
+
+        external_cache_dir = pathlib.Path(cache_dir).resolve()
```
Comment (sk4zuzu) [inline tests/integration/test_init_minimal_cluster_with_applications.py:133]:
```suggestion
        external_cache_dir = pathlib.Path(cache_dir_env).resolve()
```

File: tests/integration/test_init_minimal_cluster_with_applications.py [hunk 7]
```diff
@@ -128,8 +128,13 @@ def test_init_minimal_cluster_with_applications():
...
+
+        # Fix for Windows to avoid path like "/workdir/C:/github/m-generic-epiphany-on-aks/.cache"
+        if not external_cache_dir.exists() and pathlib.PureWindowsPath(cache_dir).is_absolute():
```
Comment (sk4zuzu) [inline tests/integration/test_init_minimal_cluster_with_applications.py:136]:
```suggestion
        if not external_cache_dir.exists() and pathlib.PureWindowsPath(cache_dir_env).is_absolute():
```

File: tests/integration/test_init_minimal_cluster_with_applications.py [hunk 8]
```diff
@@ -128,8 +128,13 @@ def test_init_minimal_cluster_with_applications():
...
+            external_cache_dir = pathlib.Path(cache_dir)
```
Comment (sk4zuzu) [inline tests/integration/test_init_minimal_cluster_with_applications.py:137]:
```suggestion
            external_cache_dir = pathlib.Path(cache_dir_env)
```

**Review by sk4zuzu — CHANGES_REQUESTED** [review]
:thinking:

**Review by sk4zuzu — APPROVED** [review]
:monkey:

**Review by erzetpe — APPROVED** [review]

**Pull request merged**